infra/main.tf

```
@@ -672,6 +703,17 @@ resource "google_storage_bucket" "batch_logs" {
   storage_class = var.batch_logs_bucket_storage_class
 }
 
+resource "random_id" "hail_query_bucket_name_suffix" {
+  byte_length = 2
+}
+
+resource "google_storage_bucket" "hail_query" {
+  name = "hail-query-${random_id.hail_query_bucket_name_suffix.hex}"
+  location = var.hail_query_bucket_location
+  force_destroy = true
+  storage_class = var.hail_query_bucket_storage_class
```

Thanks very much! I've applied these Terraform changes in our deployment.

Can we use uniform bucket permissions? This seems to be the current GCP recommendation.

```suggestion
  storage_class = var.batch_logs_bucket_storage_class
  uniform_bucket_level_access = true
}

resource "random_id" "hail_query_bucket_name_suffix" {
  byte_length = 2
}

resource "google_storage_bucket" "hail_query" {
  name = "hail-query-${random_id.hail_query_bucket_name_suffix.hex}"
  location = var.hail_query_bucket_location
  force_destroy = true
  storage_class = var.hail_query_bucket_storage_class
  uniform_bucket_level_access = true
```

---

Certainly!